iModelCore/ECPresentation/Source/Shared/Queries/CustomFunctions.cpp

```diff
@@ -424,21 +424,30 @@ struct ECExpressionScalarCacheKey
 * - Expression
 * @bsiclass
 +===============+===============+===============+===============+===============+======*/
-struct EvaluateECExpressionScalar : CachingScalarFunction<bmap<ECExpressionScalarCacheKey, std::shared_ptr<Utf8String>>>
+struct EvaluateECExpressionScalar : CachingScalarFunction<bmap<ECExpressionScalarCacheKey, std::shared_ptr<ECValue>>>
     {
     EvaluateECExpressionScalar(CustomFunctionsManager const& manager)
-        : CachingScalarFunction(FUNCTION_NAME_EvaluateECExpression, 3, DbValueType::TextVal, manager)
+        : CachingScalarFunction(FUNCTION_NAME_EvaluateECExpression, 4, DbValueType::NullVal, manager)
         {}
     void _ComputeValue(BeSQLite::DbFunction::Context& ctx, int nArgs, BeSQLite::DbValue* args) override
         {
-        ARGUMENTS_COUNT_PRECONDITION(3);
+        ARGUMENTS_COUNT_PRECONDITION(4);
 
         ECClassId classId = args[0].GetValueId<ECClassId>();
         ECInstanceId instanceId = args[1].GetValueId<ECInstanceId>();
         Utf8CP expression = args[2].GetValueText();
+        Utf8CP requestedTypeString = args[3].GetValueText();
         ECExpressionScalarCacheKey key = {classId, instanceId, expression};
 
         auto iter = GetCache().find(key);
+        PrimitiveType requestedTypePrimitive;
+        ECObjectsStatus parseStatus = ValueHelpers::ParsePrimitiveType(requestedTypePrimitive, requestedTypeString);

```

Comment:
Why not pass-in the type in the form of `PrimitiveType` instead, to avoid parsing this from string?

Reply:
Changed to pass PrimitiveType